### PR TITLE
Addition of a MCP3008 ADC Provider

### PR DESCRIPTION
**Pull request by DavidShoe**

### CODE REVIEW

File: ADC/AdcProviders.sln [hunk 1]
```diff
@@ -7,16 +7,21 @@ Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "SampleAdcConsumer", "Sample
 EndProject
 Project("{8BC9CEB8-8B4A-11D0-8D11-00A0C91BC942}") = "AdcAds1x15", "AdcAds1x15\AdcAds1x15.vcxproj", "{E62BF233-E120-4EE1-8914-EE0FE80FAD90}"
 EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "AdcMcp3008", "AdcMcp3008\AdcMcp3008.csproj", "{E187FF50-C31B-4E99-9D3A-355501E90257}"
+EndProject
 Global
 	GlobalSection(SolutionConfigurationPlatforms) = preSolution
+		Debug|Any CPU = Debug|Any CPU
```
Comment (bfjelds) [inline ADC/AdcProviders.sln]: get rid of any cpu

File: ADC/AdcMcp3008/AdcMcp3008.csproj [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+    <TargetPlatformIdentifier>UAP</TargetPlatformIdentifier>
+    <TargetPlatformVersion>10.0.10240.0</TargetPlatformVersion>
+    <TargetPlatformMinVersion>10.0.10240.0</TargetPlatformMinVersion>
+    <MinimumVisualStudioVersion>14</MinimumVisualStudioVersion>
+    <FileAlignment>512</FileAlignment>
+    <ProjectTypeGuids>{A5A43C5B-DE2A-4C0C-9213-0A381AF9435A};{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}</ProjectTypeGuids>
+    <AllowCrossPlatformRetargeting>false</AllowCrossPlatformRetargeting>
+  </PropertyGroup>
+  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' ">
+    <PlatformTarget>AnyCPU</PlatformTarget>
```
Comment (bfjelds) [inline ADC/AdcMcp3008/AdcMcp3008.csproj]: get rid of anycpu references

File: ADC/AdcMcp3008/AdcMcp3008ControllerProvider.cs [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+        {
+            return (channelMode == ProviderAdcChannelMode.SingleEnded || channelMode == ProviderAdcChannelMode.Differential);
+        }
+
+        public int ReadValue(int channelNumber)
+        {
+            byte command = (byte)channelNumber;
+            if (channelMode == ProviderAdcChannelMode.SingleEnded)
+            {
+                command |= MCP3008_SingleEnded; ;
```
Comment (bfjelds) [inline ADC/AdcMcp3008/AdcMcp3008ControllerProvider.cs]: extra semi colon: ; ;

File: ADC/AdcMcp3008/AdcMcp3008Provider.cs [hunk 4]
```diff
@@ -0,0 +1,34 @@
+﻿using System.Collections.Generic;
+using Windows.Devices.Adc.Provider;
+
+namespace AdcMcp3008
+{
+    public sealed class AdcMcp3008Provider : IAdcProvider
+    {
+        AdcMcp3008Provider()
```
Comment (bfjelds) [inline ADC/AdcMcp3008/AdcMcp3008Provider.cs]: don't need to provide an empty constructor

**Comment by ooeygui** [comment]
@jessekaplan should also take a look at this

**Comment by jessekaplan** [comment]
Thanks, Brian already sent it my way. Looks good on my end.

**Pull request merged**